adam-core/src/main/scala/org/bdgenomics/adam/rdd/settheory/Closest.scala

```diff
@@ -0,0 +1,190 @@
+package org.bdgenomics.adam.rdd.settheory
+
+import org.apache.spark.rdd.RDD
+import org.bdgenomics.adam.models.ReferenceRegion
+import org.bdgenomics.adam.rdd.ManualRegionPartitioner
+import org.bdgenomics.utils.interval.array.IntervalArray
+import scala.reflect.ClassTag
+
+/**
+ * A trait describing closest implementations that are based on sort-merge.
+ *
+ * @tparam T The type of the left records.
+ * @tparam U The type of the right records.
+ * @tparam RT The resulting type of the left after the join.
+ * @tparam RU The resulting type of the right after the join.
+ */
+sealed abstract class Closest[T: ClassTag, U: ClassTag, RT, RU]
+    extends SetTheoryBetweenCollections[T, U, RT, RU]
+    with SetTheoryPrimitive {
+
+  var currentClosest: ReferenceRegion = ReferenceRegion.empty
```

Comment:
It hasn't been a problem yet, but I am working on a fix for this in the case that the Spark Serialization rules change. I don't like this here either.